File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& arr) {
        int mn=arr[0],res=0;
        
        for(int i:arr){
            res=max(res,(i-mn));
            mn=min(mn,i);
        }
        
        return res;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int len=prices.length;
        int res=0;
        int min=prices[0];
        
        for(int i=0;i<len;++i){
            min=prices[i]<min?prices[i]:min;
            res=prices[i]-min>res?prices[i]-min:res;
        }
        
        return res;
    }
}